openmc/deplete/operator.py

```diff
@@ -251,6 +276,42 @@ def _set_number_from_mat(self, mat):
             number = density * 1.0e24
             self.number.set_atom_density(mat_id, nuclide, number)
 
+    def _set_number_from_results(self, mat, prev_res):
+        """Extracts material nuclides and number densities.
+
+        If the nuclide concentration's evolution is tracked, the densities come
+        from depletion results. Else, densities are extracted from the geometry
+        in the summary.
+
+        Parameters
+        ----------
+        mat : openmc.Material
+            The material to read from
+        prev_res : ResultsList
+            Results from a previous depletion calculation
+
+        """
+        mat_id = str(mat.id)
+
+        # Get nuclide lists from geometry and depletion results
+        depl_nuc = prev_res[-1].nuc_to_ind.keys()
+        geom_nuc_densities = mat.get_nuclide_atom_densities()
+        geom_nuc = [x[0] for x in list(geom_nuc_densities.values())]
```

Comment:
You can use a set comprehension here and then change `set(geom_nuc)` -> `geom_nuc` below. Also, there's no need for an explicit conversion to `list` since `values()` is iterable. So, it would be
```Python
geom_nuc = {x[0] for x in geom_nuc_densities.values()}
```